apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts

```diff
@@ -0,0 +1,95 @@
+import { EnvironmentRepository, IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { providers } from '@novu/shared';
+
+export const ENVIRONMENT_NAME_TO_SHORT_NAME = { ['Development']: 'dev', ['Production']: 'prod', ['undefined']: '' };
+
+export async function addIntegrationIdentifierMigrationBatched() {
+  // eslint-disable-next-line no-console
+  console.log('start migration - add integration identifier migration');
+
+  const integrationRepository = new IntegrationRepository();
+  const environmentRepository = new EnvironmentRepository();
+  const batchSize = 500;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  for await (const integration of integrationRepository.findBatch({} as any, '', {}, batchSize)) {
+    // eslint-disable-next-line no-console
+    console.log(`integration ${integration._id}`);
+
+    const updatePayload = await getUpdatePayload(integration, environmentRepository);
+
+    await integrationRepository.update(
+      {
+        _id: integration._id,
+        _environmentId: integration._environmentId,
+        _organizationId: integration._organizationId,
+      },
+      {
+        $set: updatePayload,
+      }
+    );
+    // eslint-disable-next-line no-console
+    console.log(`integration ${integration._id} - name & identifier updated`);
+  }
+  // eslint-disable-next-line no-console
+  console.log('end migration');
+}
+
+export async function addIntegrationIdentifierMigration() {
+  // eslint-disable-next-line no-console
+  console.log('start migration - add integration identifier migration');
+
+  const integrationRepository = new IntegrationRepository();
+  const environmentRepository = new EnvironmentRepository();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const integrations = await integrationRepository.find({} as any);
+
+  for (const integration of integrations) {
+    // eslint-disable-next-line no-console
+    console.log(`integration ${integration._id}`);
+
+    const updatePayload = await getUpdatePayload(integration, environmentRepository);
+
+    await integrationRepository.update(
+      {
+        _id: integration._id,
+        _environmentId: integration._environmentId,
+        _organizationId: integration._organizationId,
+      },
+      {
+        $set: updatePayload,
+      }
+    );
+    // eslint-disable-next-line no-console
+    console.log(`integration ${integration._id} - name & identifier updated`);
+  }
+  // eslint-disable-next-line no-console
+  console.log('end migration');
+}
+
+async function getUpdatePayload(integration: IntegrationEntity, environmentRepo: EnvironmentRepository) {
+  const updatePayload: Partial<IntegrationEntity> = {};
+
+  if (!integration.name) {
+    updatePayload.name = getDisplayName(integration.providerId);
+  }
+
+  if (!integration.identifier) {
+    updatePayload.identifier = await getIntegrationIdentifier(environmentRepo, integration);
+  }
```

Comment:
This is how I'm going to generate these for the create endpoint, maybe you could do it similarly:
```
const providerIdCapitalized = `${command.providerId.charAt(0).toUpperCase()}${command.providerId.slice(1)}`;
const defaultName = providers.find((provider) => provider.id === command.providerId)?.displayName ?? providerIdCapitalized;
const name = command.name ?? defaultName;
const identifier = command.identifier ?? `${slugify(name, { lower: true, strict: true })}-${shortid.generate()}`;
```

Reply:
It would be great if both places (migration and use case) share the same function, in my opinion. 👍🏻

Reply:
> This is how I'm going to generate these for the create endpoint, maybe you could do it similarly:
> 
> ```
> const providerIdCapitalized = `${command.providerId.charAt(0).toUpperCase()}${command.providerId.slice(1)}`;
> const defaultName = providers.find((provider) => provider.id === command.providerId)?.displayName ?? providerIdCapitalized;
> const name = command.name ?? defaultName;
> const identifier = command.identifier ?? `${slugify(name, { lower: true, strict: true })}-${shortid.generate()}`;
> ```

Is this logic already merged with next? If not can you pin here the relevant PR id so I will know where it is, once it is merged i will reuse the same function logic here

Reply:
the PR: https://github.com/novuhq/novu/pull/3674

Reply:
in order to not wait for the merge to next i committed duplication of this logic for now.